risc0/zkvm/sdk/rust/guest/src/env.rs

```diff
@@ -128,6 +128,17 @@ pub fn get_cycle_count() -> usize {
     ENV.get().get_cycle_count()
 }
 
+/// Print a message to the debug console.
+pub fn println(msg: &str) {
+    let desc = IoDescriptor {
+        size: msg.len() as u32,
+        addr: msg.as_ptr() as u32,
+    };
+    let ptr: *const IoDescriptor = &desc;
+    memory_barrier(ptr);
+    unsafe { GPIO_LOG.as_ptr().write_volatile(&desc) };
+}
+
```

Comment:
OK, mind if I do this in a followup PR?

Reply:
WDYT about renaming `println` to `debug`, or perhaps `console_log`

Reply:
I was actually thinking of having the `log` crate to work in `no_std` and have it use this GPIO

Reply:
I tried the `log!` macro and ran into trouble dealing with non-literal strings. For now, I'll just revert the GPIO_LOG impl to be null terminated and use `format!("{}\0", msg)`. This is pretty slow in the guest, but I'm hoping we can come up with a better solution later.

One idea is to have GPIO_LOG *and* GPIO_LOG_NULL. Since in C++ it would be expensive to compute `strlen` in the guest, but it's also expensive to concat a NULL terminator in rust.